src/TorchSharp/Utils/tensorboard/Utils.cs

```diff
@@ -0,0 +1,125 @@
+                        long H = I.shape[2];
+                        long W = I.shape[3];
+                        ncols = Math.Min(ncols, nimg);
+                        long nrows = (long)Math.Ceiling((double)nimg / (double)ncols);
+                        Tensor canvas = zeros(new long[] { 3, H * nrows, W * ncols }, I.dtype);
+                        long i = 0;
+                        for (long r = 0; r < nrows; r++) {
+                            for (long c = 0; c < ncols; c++) {
+                                if (i >= nimg)
+                                    break;
+                                canvas.narrow(1, r * H, H).narrow(2, c * W, W).copy_(I[i]);
+                                i++;
+                            }
+                        }
+                        return canvas;
+                    }
+
+                    /// <summary>
+                    /// https://github.com/pytorch/pytorch/blob/6c30dc6ceed5542351b3be4f8043b28020f93f3a/torch/utils/tensorboard/_utils.py#L95
+                    /// </summary>
+                    /// <param name="tensor"> Image data </param>
+                    /// <param name="input_format"> Image data format specification of the form NCHW, NHWC, CHW, HWC, HW, WH, etc. </param>
+                    /// <returns></returns>
+                    public static Tensor convert_to_HWC(Tensor tensor, string input_format)
+                    {
+                        Trace.Assert(tensor.shape.Length == input_format.Length, $"size of input tensor and input format are different. tensor shape: ({string.Join(", ", tensor.shape)}), input_format: {input_format}");
+                        input_format = input_format.ToUpper();
+
+                        if (input_format.Length == 4) {
+                            long[] index = "NCHW".Select(c => Convert.ToInt64(input_format.IndexOf(c))).ToArray();
+                            Tensor tensor_NCHW = tensor.permute(index);
+                            Tensor tensor_CHW = make_grid(tensor_NCHW);
+                            return tensor_CHW.permute(1, 2, 0);
+                        }
+
+                        if (input_format.Length == 3) {
+                            long[] index = "HWC".Select(c => Convert.ToInt64(input_format.IndexOf(c))).ToArray();
+                            Tensor tensor_HWC = tensor.permute(index);
+                            if (tensor_HWC.shape[2] == 1)
+                                tensor_HWC = tensor_HWC.expand(-1, -1, 3);
+                            return tensor_HWC;
+                        }
+
+                        if (input_format.Length == 2) {
+                            long[] index = "HW".Select(c => Convert.ToInt64(input_format.IndexOf(c))).ToArray();
+                            tensor = tensor.permute(index);
+                            tensor = tensor.unsqueeze(-1).expand(-1, -1, 3);
+                            return tensor;
+                        }
```

Comment:
> Also, if I may -- I think the TensorToSKBitmap() function in Summary.cs should handle grayscale images (C == 1) by choosing 'Gray8' as the color type. Currently, the code won't handle grayscale images at all, it seems.

This is how pytorch handles the case of C == 1
https://github.com/pytorch/pytorch/blob/master/torch/utils/tensorboard/_utils.py#L119-L120
https://github.com/pytorch/pytorch/blob/master/torch/utils/tensorboard/_utils.py#L123-L127